### PR TITLE
[Android] Center elements in CarouselView

### LINKED ISSUE #8225: [Bug] CarouselView not working in ios application
**Description**
I had created mobile application using Xamarin.Forms(version 4.3.0.908675) and use MVVM pattern. I am using CarouselView in one of my page. as below 

`<CarouselView ItemsSource="{Binding SelectedProduct.ProductImages}" Position="{Binding SliderPosition}" PositionChangedCommand="{Binding PositionSelectedCommand}" PositionChangedCommandParameter="{Binding SliderPosition}">
                                        <CarouselView.ItemTemplate>
                                            <DataTemplate x:DataType="local:ProductImage">
                                                <Grid>
                                                    <Label Text="ProductImages" x:Name="productImage" IsVisible="False" />
                                                    <ffimageloading:CachedImage LoadingPlaceholder="img_loading.png" ErrorPlaceholder="img_error.png" DownsampleToViewSize="True" Source="{Binding FilePath, Converter={StaticResource ImagePathConverter},ConverterParameter={x:Reference productImage}}" Aspect="AspectFill" />
                                                    <StackLayout BackgroundColor="{StaticResource CarouselHeaderBackGroundColor}" Padding="10" VerticalOptions="EndAndExpand" IsVisible="{Binding Description, Converter={StaticResource StringToBoolConverter}}">
                                                        <Label TextColor="White" Text="{Binding Description}" FontSize="{StaticResource SmallFontSize}" HorizontalOptions="Center" VerticalOptions="EndAndExpand"/>
                                                    </StackLayout>
                                                </Grid>
                                            </DataTemplate>
                                        </CarouselView.ItemTemplate>
                                    </CarouselView>`

The page is working fine in android devices but not working in ios devices (Did testing in iPhone 7s iOS 12.2 and simulator iPhone X iOS 12.2).

**Expected Behavior**
CarouselView control must work in iOS devices

**Actual Behavior**
CarouselView control not working in iOS devices

**Basic Information**
Version with issue: Xamarin.Forms 4.3.0.908675
IDE: Microsoft Visual Studio Community 2017
Platform Target Frameworks:
Android: Android 9.0 (API Level 28 - Pie)
iOS: iOS 12.2

### LINKED ISSUE #8638: [Bug] CarouselView inconsistent layout positioning
### Description

I have the following layout using the `PeekAreaInsets`:

```
<CarouselView 
            PeekAreaInsets="40"
            ItemsSource="{StaticResource Banners}" 
            Position="3">
            <CarouselView.ItemsLayout>
                <LinearItemsLayout Orientation="Horizontal"
                                   SnapPointsAlignment="Center"
                                   SnapPointsType="MandatorySingle"
                                   ItemSpacing="8"/>
            </CarouselView.ItemsLayout>
            <CarouselView.ItemTemplate>
                <DataTemplate>
                    <Image Source="{Binding .}"/>
                </DataTemplate>
            </CarouselView.ItemTemplate>
        </CarouselView>
```

### Expected Behavior

The centered item is always positioned in the center.

### Actual Behavior

On the first item and the last item, the layout differs. See images.

I get why the layout is doing this as there is no item to consume the 40 inset and so I only get an 8 padding (if I'm interpreting correctly). How can I achieve the consistent layout I want?

Slightly related, what I ACTUALLY want is an carousel that loops, and thus there would ALWAYS be a left and right, prev and next, image. I'll file another issue to track that.

### Basic Information

- Version with issue: 4.4.0-pre3
- Platform Target Frameworks: 
  - Android: 10

### Screenshots

Centered as expected:

image

First item, not centered. The margins on the left and right should be consistent with the above image. I expect the positioning to be consistent no matter what item I'm on. There should be 40 + 8 padding to the left.

image

Last item, not centered. Again, I expect consistent layout. There should be 40 + 8 padding to the right.

image

### Reproduction Link

https://github.com/davidortinau/FlyMe/blob/bug/carousel_peek_layout/FlyMe/FlyMe/Views/MyFlightsPage.xaml#L18

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

In this PR we have two fixes related to the `CarouselView` and element positioning:
- When using `PeekAreaInsets` the result between Android and iOS was different. This was because in Android we did not convert the peek insets values to Pixels.
- While in iOS we always show the elements in the center, it was not the case on Android. Applied changes in `CarouselSpacingItemDecoration` to fix this issue.

carousel-centerered

### Issues Resolved ### 

- fixes #8225 (It does not correct the issue related to `IsVisible` but the size item issue of the attached sample using PeekAreaInsets)
- fixes #8638

### API Changes ###
 
 None

### Platforms Affected ### 

- Android

### Behavioral/Visual Changes ###

None

#### After
issue8525-vertical-after
issue8525-horizontal-after

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

**Comment by samhouts** [comment]
@jsuarezruiz Please retarget to 4.4.0. Thanks!

**Comment by jsuarezruiz** [comment]
Retargeted to 4.4.0.

**Comment by jsuarezruiz** [comment]
If we look for the opposite behavior, that is, have the same behavior in iOS that we have in Android now (do not center and add spaces at the beginning and end) we must apply changes in `CarouselViewLayout`. In this Gist https://gist.github.com/jsuarezruiz/6cbb45d7a85ec6dfc415ef625cfbf2ee are the changes that would be necessary.

File: Xamarin.Forms.Platform.Android/CollectionView/RecyclerViewScrollListener.cs [hunk 1]
```diff
@@ -76,31 +74,28 @@ public override void OnScrolled(RecyclerView recyclerView, int dx, int dy)
 			}
 		}
 
-		static int CalculateCenterItemIndex(int firstVisibleItemIndex, int lastVisibleItemIndex, LinearLayoutManager linearLayoutManager)
+		static int CalculateCenterItemIndex(int firstVisibleItemIndex, RecyclerView recyclerView, LinearLayoutManager linearLayoutManager)
```
Comment (rmarinho) [inline Xamarin.Forms.Platform.Android/CollectionView/RecyclerViewScrollListener.cs]: we need to pass the firstvfisibleItemIndex? we can grab it from linearLayoutManager i think

**Comment by samhouts** [comment]
I did not run tests. We'll use the release test run to validate this PR.

**Pull request merged**